checker.cpp

```diff
@@ -1,46 +1,70 @@
-#include <iostream>
-#include <assert.h>
-#include <unistd.h>
+#include "checker.h"
+
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+int VitalsMonitor::vitalsOk(float temperature, float pulseRate, float spo2) {
+    int status = VitalsMonitor::bodyTemperatureOk(temperature);
+    if (0 != status) {
+        status = VitalsMonitor::pulseRateOk(pulseRate);
+        if(0 != status) {
+            status = VitalsMonitor::oxygenSaturationOk(spo2);
+        }
     }
+    return status;
+}
 
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+int VitalsMonitor::bodyTemperatureOk(float temperature) {
+    int status = 1;
+    if(VitalsMonitor::isBodyTemperatureNotOk(temperature)) {
```

Comment:
Try naming functions without "Not" in them, to avoid double-negative / non-intuitive structures.
E.g., entering the if statement (true condition) and returning false (0) status.
How about `bodyTemperatureNeedsAttention`?